docs/generating-a-provider.md

```diff
@@ -0,0 +1,391 @@
+           "github.com/crossplane-contrib/terrajet/pkg/types/conversion"
+       )
+       
+       resourceMap := conversion.GetV2ResourceMap(tf.Provider())
+       ```
+
+       And in `go.mod` file, set the following `replace directive`
+       (uncomment related section):
+
+       ```
+       github.com/hashicorp/terraform-plugin-sdk => github.com/turkenh/terraform-plugin-sdk v1.17.2-patch1
+       ```
+
+       Run:
+       ```bash
+       go mod tidy
+       ```
+
+5. Implement `ProviderConfig` logic. In `provider-tf-template`, there is already
+   a boilerplate code in file `internal/clients/${ProviderNameLower}.go` which
+   takes care of properly fetching secret data referenced from `ProviderConfig`
+   resource.
+
+   For our GitHub provider, we need to check [Terraform documentation for provider
+   configuration] and provide the keys there:
+
+   ```go
+   const (
+     keyBaseURL = "base_url"
+     keyOwner = "owner"
+     keyToken = "token"
+
+     // GitHub credentials environment variable names
+     envToken = "GITHUB_TOKEN"
+   )
+
+   func TerraformSetupBuilder(version, providerSource, providerVersion string) terraform.SetupFn {
+     ...
+     // set provider configuration
+     ps.Configuration = map[string]interface{}{}
+     if v, ok := githubCreds[keyBaseURL]; ok {
+         ps.Configuration[keyBaseURL] = v
+     }
+     if v, ok := githubCreds[keyOwner]; ok {
+         ps.Configuration[keyOwner] = v
+     }
+     // set environment variables for sensitive provider configuration
+     ps.Env = []string{
+         fmt.Sprintf(fmtEnvVar, envToken, githubCreds[keyToken]),
+     }
+     return ps, nil
+   }
+   ```
+
+6. Before generating all resources that the provider has, let's go step by step
+   and only start with generating `github_repository` and `github_branch`
+   resources.
+
+   To limit the resources to be generated, we need to provide an include list
+   option with `tjconfig.WithIncludeList` in file `config/provider.go`:
+
+   ```go
+   pc := tjconfig.NewProvider(resourceMap, resourcePrefix, modulePath,
+       tjconfig.WithDefaultResourceFn(defaultResourceFn),
+       tjconfig.WithIncludeList([]string{
+           "github_repository$",
+           "github_branch$",
+       }))
+   ```
+
+7. Finally, we would need to add some custom configurations for these two
+   resources as follows:
+
+   ```bash
+   # Create custom configuration directory for whole repository group
+   mkdir config/repository
+   # Create custom configuration directory for whole branch group
+   mkdir config/branch
+   ```
+
+   ```bash
+   cat <<EOF > config/repository/config.go
+   package repository
+
+   import "github.com/crossplane-contrib/terrajet/pkg/config"
+
+   func Customize(p *config.Provider) {
+       p.AddResourceConfigurator("github_repository", func(r *config.Resource) {
+   
+           // we need to override the default group that terrajet generated for
+           // this resource, which would be "github"  
```

Comment:
Why would the generated group be `github` rather than the last part of any `_` string?

Reply:
Terrajet is doing best in terms of assigning a group, but it is always easily possible to override them. This is the best default based on our experience so far, and changing it would be a bit tricky as it _might_ break existing providers.